sys/include/net/ng_llhdr.h

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (C) 2015 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     net
+ * @{
+ *
+ * @file
+ * @brief       Generic link layer header
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ */
+
+#ifndef LLHDR_H_
+#define LLHDR_H_
+
+#include <string.h>
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Generic link layer header
+ *
+ * The actual addresses are put into the memory following this datastructure.
+ */
+typedef struct __attribute__((packed)) {
+    uint8_t addr_len;           /**< link layer address length in byte */
```

Comment:
If we only support one address length we will not be complaiant to most 6LoWPAN implementations. There must be a `src_len` and `dest_len` field to support this (in IEEE 802.15.4 the length of the source address might differ from the length of the destination address)